src/components/editor/metaform-editor-right-drawer-visibility.tsx

```diff
@@ -0,0 +1,338 @@
+              { pendingForm.sections!.map(section => section.fields!.map((field, index) => {
+                const constructedKey = `${field.title}-${index}`;
+                if (field.type === "radio" || field.type === "checklist" || field.type === "boolean") {
+                  if (field.name !== selectedComponent) {
+                    return (
+                      <MenuItem value={ field.name } key={ constructedKey }>
+                        { field.title }
+                      </MenuItem>
+                    );
+                  }
+                }
+                return null;
+              }))
+              }
+            </Select>
+          </FormControl>
+        </>
+      );
+    }
+  };
+
+  /**
+   * 
+   * Render field Condition info component
+   */
+  const fieldConditionInfoComponent = () => {
+    if (requiredConditionField === "" && disabledValue) {
+      return (
+        <Typography sx={{ color: "gray" }}>
+          { strings.draftEditorScreen.editor.visibility.conditionalFieldInfo }
+        </Typography>
+      );
+    }
+
+    return null;
+  };
+
+  /**
+   * Check options of demanded metaformField 
+   * @param field MetaformField where we get option values
+   * @returns options from field
+   */
+  const renderFieldConditionOptions = (field: MetaformField) => (
+    field.options!.map((option, index) => {
+      const constructedKey = `${option.text}-${index}-`;
+      return (
+        <MenuItem value={ option.name } key={ constructedKey }>
+          { option.text }
+        </MenuItem>
+      );
+    })
+  );
+
+  /**
+   * Render visiblity condition select menu
+   * 
+   */
+  const conditionValueField = () => {
+    if (sectionIndex !== undefined) {
+      if (!requiredConditionField) {
+        return null;
+      }
+      return (
+        <>
+          <FormControl fullWidth>
+            <InputLabel id="visiblityConditionLabel">{ strings.draftEditorScreen.editor.visibility.conditionalFieldValue }</InputLabel>
+            <Select
+              fullWidth
+              labelId="visiblityConditionLabel"
+              label={ strings.draftEditorScreen.editor.visibility.conditionalFieldValue }
+              value={ requiredConditionEqualsValue }
+              onChange={ event => updateFormFieldOrSectionVisiblityValues({
+                ...metaFormFieldSection,
+                visibleIf: {
+                  field: requiredConditionField,
+                  equals: event.target.value,
+                  notEquals: "",
+                  and: [],
+                  or: []
+                }
+              }, event.target.value)
+              }
+            >
+              { pendingForm.sections!.map(section => section.fields!.map(field => {
+                if (field.name === requiredConditionField) {
+                  return (
+                    renderFieldConditionOptions(field)
+                  );
+                }
+                return null;
```

Comment:
This return should be redundant.

Reply:
Array.prototype.map() expects a value to be returned at the end of arrow function